pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1368,7 +1368,7 @@ func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta
 	// Iterate over topology pairs to get any of the pods being affected by
 	// the scheduled pod anti-affinity terms
 	for topologyKey, topologyValue := range node.Labels {
-		if topologyMaps.topologyPairToPods[topologyPair{key: topologyKey, value: topologyValue}] != nil {
+		if topologyMap[topologyPair{key: topologyKey, value: topologyValue}] != 0 {
```

Comment:
A nit: given that we're checking `!= 0` here, it seems we don't need the logic of "checking if it's == 0 and then delete that entry" during the construction?

Or, just use `if _, ok := topologyMap[topologyPair{key: topologyKey, value: topologyValue}]; !ok` to keep the logic consistent with L1392?

Reply:
how about we change L1392 to check for ```== 0``` instead of existence? we can remove the logic that delete the entry in a followup PR that converts the type to *int64 and fixes the test at the same time (tests that use DeepEqual)

Reply:
SGTM.